### PR TITLE
#162162967 Users should be able to create articles

### PR DESCRIPTION
**Pull request by Luleherll**
**What does this PR do?**
- Add redux thunks for getting and posting data
- Create article reducer function

**Description of the tasks to be completed?**
- Update the `types` file in the actions folder.
- Modify `ArticleReducer` file in reducers file
- Create rich text editor
- Create article components

**How to manually test it**
- Clone the repository by running `git clone https://github.com/andela/ah-frontend-poseidon.git`
- Run `git checkout ft-create-articles-162162967`
- Start the server by running `yarn start`

**Screenshots**
screen shot 2019-01-23 at 6 53 01 pm
screen shot 2019-01-23 at 6 52 27 pm
screen shot 2019-01-23 at 6 53 10 pm

**What are the relevant pivotal tracker stories?**
#162162967

### CODE REVIEW

**Review by arthurarty — APPROVED** [review]

**Review by arthurarty — APPROVED** [review]

**Comment by katunold** [comment]
Well done @Luleherll, Please squash your commits

File: src/components/articles/Editor.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        id='contentArea'
+        ></div>
+        <div className='col'>
+          <button disabled={disabled} className={`basic-button${disable}`} id='post' type='submit'>Post</button>
+        </div>
+        </div>
+      </div>
+    )
+  }
+}
```
Comment (katunold) [inline src/components/articles/Editor.jsx]: @Luleherll, please add an empty line at the end of this file

File: src/components/articles/Tags.jsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+
+export default ({tags})=>{
+    return(
+      <div>
+        <i className="fa fa-tags">Tags: </i>
+        {tags.map((tag,i) =><a key={i}><span key={i} className='badge badge-info'>{tag}</span></a>)}
+      </div>
+    )
+}
```
Comment (katunold) [inline src/components/articles/Tags.jsx]: @Luleherll, please add an empty line at the end of this file too

File: src/components/articles/__tests__/EditorTest.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    wrapper.find('#command').first().simulate('click');
+  });
+  it('should trigger horizontalRule onClick event', () => {
+    wrapper.find('#horizontalRule').first().simulate('click');
+  });
+
+  it('Editor snapshot', () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+});
```
Comment (katunold) [inline src/components/articles/__tests__/EditorTest.js]: @Luleherll, please add an empty line at the end of this file

File: src/components/dashboard/ArticleComponent.jsx [hunk 4]
```diff
@@ -43,4 +45,4 @@ Article.propTypes = {
   }).isRequired,
 };
 
-export default Article;
+export default Article;
```
Comment (katunold) [inline src/components/dashboard/ArticleComponent.jsx]: @Luleherll, please add an empty line at the end of this file

File: src/components/dashboard/Dashboard.scss [hunk 5]
```diff
@@ -92,4 +85,4 @@ p {
 
 div.article-row { 
     padding: 6px;
-}
+}
```
Comment (katunold) [inline src/components/dashboard/Dashboard.scss]: @Luleherll, please add an empty line at the end of this file

File: src/components/dashboard/NavBarComponent.jsx [hunk 6]
```diff
@@ -34,4 +33,4 @@ const NavBar = () => {
   );
 };
 
-export default NavBar;
+export default NavBar;
```
Comment (katunold) [inline src/components/dashboard/NavBarComponent.jsx]: @Luleherll, please add an empty line at the end of this file

File: src/components/dashboard/SideBarComponent.jsx [hunk 7]
```diff
@@ -36,16 +36,16 @@ SideBar.propTypes = {
...
+      favourites_count: PropTypes.number.isRequired,
       author: PropTypes.object.isRequired,
       id: PropTypes.number.isRequired,
     }),
   ).isRequired,
 };
 
 
-export default SideBar;
+export default SideBar
```
Comment (katunold) [inline src/components/dashboard/SideBarComponent.jsx]: @Luleherll, please add an empty line at the end of this file

File: src/redux/thunks/__tests__/ThunksTest.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  it('Should handle ERROR_OCCURRED', () => {
+
+    httpMock.onPost('articles/').reply(403, error);
+    store
+      .dispatch(postDataThunk('articles/', data, createArticle, 'put'))
+      .then(() => {
+        expect(store.getActions()).toEqual([{ type: ERROR_OCCURRED, error }]);
+      });
+  });
+});
```
Comment (katunold) [inline src/redux/thunks/__tests__/ThunksTest.js]: @Luleherll, please add an empty line at the end of this file

File: src/views/homeView.jsx [hunk 9]
```diff
@@ -1,17 +1,88 @@
...
-export default HomeView;
+const mapStateToProps = ({ articles: { articles }}, props) =>({
+  articles: articles,
+})
+
+const mapDispatchToProps = dispatch => ({
+  actions: bindActionCreators({ getOneArticle, getDataThunk }, dispatch)
+  })
+
+export default connect(mapStateToProps, mapDispatchToProps)(HomeView)
```
Comment (katunold) [inline src/views/homeView.jsx]: @Luleherll, please add an empty line at the end of this file

**Review by katunold — CHANGES_REQUESTED** [review]

File: src/components/articles/Articles.jsx [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+const mapStateToProps = (state, props) =>({
+  article: state.articles.article,
+})
+
+const mapDispatchToProps = dispatch => ({
+actions: bindActionCreators({ ...Actions, getOneArticle, deleteArticle}, dispatch)
+})
+
+export default connect(mapStateToProps, mapDispatchToProps)(ArticleView)
```
Comment (katunold) [inline src/components/articles/Articles.jsx]: @Luleherll, you forgot to an empty line at the end of this file.

File: src/components/articles/articles.scss [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+}
+.commands:hover{
+  background-color: #3f51b5;
+}
+#contentArea{
+  outline: 0px;
+  height: 420px;
+  background-color: lightgray;
+  overflow: scroll;
+}
```
Comment (katunold) [inline src/components/articles/articles.scss]: This too needs an empty line at the end of the file.

**Review by katunold — CHANGES_REQUESTED** [review]

**Review by katunold — APPROVED** [review]

**Review by okellogabrielinnocent — APPROVED** [review]

**Review by billkabanga — APPROVED** [review]

**Review by aayebare — APPROVED** [review]
I've tested this. The functionality works fine. Great work Lule

**Pull request merged**